tools/codegen/api/python.py

```diff
@@ -520,12 +702,65 @@ def argument(a: Argument) -> PythonArgument:
         default_init=None,
     )
 
-def signature(f: NativeFunction, *, method: bool = False) -> PythonSignature:
+def argument_type_str_pyi(t: Type) -> str:
+    # vestige from 'dynamic_type': remove optional-ness
+    if isinstance(t, OptionalType):
+        t = t.elem
```

Comment:
BTW, this is probably wrong; if we accept something optionally, we should probably be generating `Optional[T]` instead. (No action needed here, you are byte-for-byte compat)

Reply:
I cleaned this up a little more- I was actually doing something pretty ugly before (taking away `Optional`ness inside of this function, but adding it back in `argument_str_pyi()`.

I also merged `argument_str()` and `argument_str_pyi()` together, and tried to more explicitly move pyi differences in types to this function, vs. differences in naming/defaulting to `argument_str()`.